### PR TITLE
Tune Shell evasion Rule to incorporate GTFOArgs shell evasion

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
#2686

## Summary
Tune existing Rule "Linux Restricted Shell Breakout via Linux Binary(s)" to incorporate GTFOArgs commands for shell evasion.
Scoped: GTFOArgs for Shell Evasion
Commands: awk, find, ssh, tar, zip
Refer - https://gtfoargs.github.io/#+shell
Testing Details - #2686 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/execution_shell_evasion_linux_binary.toml [hunk 1]
```diff
@@ -136,7 +136,7 @@ process where host.os.type == "linux" and event.type == "start" and
     (process.parent.name == "expect" and process.parent.args == "-c" and process.parent.args in ("spawn /bin/sh;interact", "spawn /bin/bash;interact", "spawn /bin/dash;interact", "spawn sh;interact", "spawn bash;interact", "spawn dash;interact")) or
     (process.parent.name == "mysql" and process.parent.args == "-e" and process.parent.args in ("\\!*sh", "\\!*bash", "\\!*dash", "\\!*/bin/sh", "\\!*/bin/bash", "\\!*/bin/dash")) or
     (process.parent.name == "ssh" and process.parent.args == "-o" and process.parent.args in ("ProxyCommand=;sh 0<&2 1>&2", "ProxyCommand=;bash 0<&2 1>&2", "ProxyCommand=;dash 0<&2 1>&2", "ProxyCommand=;/bin/sh 0<&2 1>&2", "ProxyCommand=;/bin/bash 0<&2 1>&2", "ProxyCommand=;/bin/dash 0<&2 1>&2")) or
-    (process.parent.name in ("nawk", "mawk", "awk", "gawk") and process.parent.args : "BEGIN {system(*)}")
```
Comment (w0rk3r) [inline rules/linux/execution_shell_evasion_linux_binary.toml:139]: > ) and process.parent.args : "BEGIN {system(*)}"

We don't need this condition anymore?

File: rules/linux/execution_shell_evasion_linux_binary.toml
Comment (w0rk3r) [inline rules/linux/execution_shell_evasion_linux_binary.toml:1]: Missing bump on updated_date

File: rules/linux/execution_shell_evasion_linux_binary.toml [hunk 2]
```diff
@@ -109,7 +109,7 @@ process where host.os.type == "linux" and event.type == "start" and
 
     /* launching shells from unusual parents or parent+arg combos */
     (process.name in ("bash", "sh", "dash","ash") and
-        (process.parent.name in ("byebug","git","ftp","strace")) or
+        (process.parent.name in ("byebug","git","ftp","strace","nawk", "mawk", "awk", "gawk", "tar", "zip")) or
 
         /* shells specified in parent args */
         /* nice rule is broken in 8.2 */
```
Comment (w0rk3r) [inline rules/linux/execution_shell_evasion_linux_binary.toml:115]:
```suggestion
```
I think we can remove this one

**Review by w0rk3r — APPROVED** [review]
LGTM

**Review by imays11 — APPROVED** [review]
Looks good!

**Pull request merged**